### PR TITLE
lint: add more linting rules for track config.json

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR adds some more linting rules for the track's config.json.

I'm not really happy with our error messages, but we can improve that later on. At this stage, it is important to just have some validation.

- refactor: create module for track config.json validation
- lint: add validator to check boolean property
- lint: add validator to check integer property
- lint: validate basic property of track config.json file
- lint: valid tags property of track config.json file
- lint: update output of applied rules

### CODE REVIEW

**Review by ee7 — APPROVED** [review]
Non-blocking comment: Can we think of a better name than `track.nim`? I think:
- it's too close to `src/sync/tracks.nim`
- it sounds like it handles linting the whole track, when actually it just handles the root `config.json` file

I'm fine with something like these:
- `track_config.nim`
- `root_config.nim`

or whatever you like.

Otherwise LGTM - approved, conditional on squashing at merge-time.

File: src/lint/track.nim [hunk 1]
```diff
@@ -38,42 +38,40 @@ let tags = @[
   "used_for/robotics",
   "used_for/scientific_calculations",
   "used_for/scripts",
-  "used_for/web_development"
-]
+  "used_for/web_development",
+].toHashSet()
```
Comment (ErikSchierboom) [inline src/lint/track.nim:42]: Doh. I should have come up with this myself 🤦

**Comment by ErikSchierboom** [comment]
I'm fine with `track_config.nim`. Can you update?

**Comment by ErikSchierboom** [comment]
Squashed and merged

**Pull request merged**